### PR TITLE
✅ Greengrocers - New Layout

### PR DESCRIPTION
**Pull request by lucianosimoni**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -215,14 +214,22 @@ function removeFromCart(item) {
...
-  const selfFloat = parseFloat(total.innerText.slice(1)) 
-  const fixedValue = Number(selfFloat + value).toFixed(2)
-  console.log(`Add value ${value} to total ${total.innerText}`)
-  
-  total.innerText = `£${fixedValue}`
+// Iterates though the Cart items checking for the item quantity.
+function updateTotal() {
+  const totalSpan = document.querySelector(".total-number")
+  let total = 0
+
```
Comment (SamuelOdigie) [inline index.js:275]: This is great work, clean code. so im learning as a read

File: index.js [hunk 2]
```diff
@@ -215,14 +214,22 @@ function removeFromCart(item) {
...
+    for (let i = 1; i <= item.quantity; i++) {
+      // Looping though the item quantity, add here
+      total += item.price
+    }
+  })
+
+  // Fix the value to only 2 zeros after .
+  total = total.toFixed(2)
+  totalSpan.innerText = `£${total}` // Updates HTML
 }
```
Comment (SamuelOdigie) [inline index.js:287]: This is great work, clean code. so im learning as a read

File: README.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+I choose to create a new layout for this challenge. The design can be seen [here](https://miro.com/welcomeonboard/ME5kS3FvVHBmZU1Sblc3VW5reXpzVTZuaXJGTUViVGZCc3dxYmVyN05DYnZUc3N0YkFnczhWS2NVSFFrRzVFM3wzNDU4NzY0NTMyNTk1NDg5MTc2fDI=?share_link_id=984467263935) *(comments are welcome 🙂)*
+
+Check it out on your browser [HERE](https://lucianosimoni.github.io/js-dom-greengrocers/)
+
+![Result](design.png)
+
+# Boolean - Greengrocers
 In this exercise we explore a common scenario in eCommerce, adding and removing items from the cart, and calculating the total. You final solution should look like this:
 
 ![Result](result.gif)
```
Comment (Joelzor) [inline README.md:12]: You are a madman. Very impressive

File: index.js [hunk 4]
```diff
@@ -1,55 +1,314 @@
...
+    // }
+  ]
+};
+
+// Store or else
+function resetList(listName) {
+  if (listName === "store") {
+    // Removes all the children of the ul
+    const ul = document.querySelector(".store--item-list")
+    while (ul.hasChildNodes()) {
```
Comment (Joelzor) [inline index.js:82]: Interesting reset function, hasChildNodes isn't something I've seen before. Very clean

File: index.js [hunk 5]
```diff
@@ -1,55 +1,314 @@
...
+
+  if (sortBy === "alpha") {
+    sortedItems.sort(compareName)
+  }
+  else if (sortBy === "price") {
+    sortedItems.sort(comparePrice)
+  }
+  else {
+    // Return the default array
+    return [...data.items]
```
Comment (Joelzor) [inline index.js:158]: Spread operator is awesome, surprised we never covered it

File: index.js [hunk 6]
```diff
@@ -1,55 +1,314 @@
...
+
+  renderCart()
+}
+
+// Iterates though the Cart items checking for the item quantity.
+function updateTotal() {
+  const totalSpan = document.querySelector(".total-number")
+  let total = 0
+
+  data.cart.forEach((item) => {
```
Comment (Joelzor) [inline index.js:276]: You could try using the awesome and confusing reduce function to do the same thing :)

**Pull request closed without merging**